Makefile

```diff
@@ -12,11 +13,22 @@ clean:
 # the "-tags daemon" part is temporary
 .PHONY: test
 test:
-	./scripts/test/unit $(shell go list ./... | grep -v /vendor/)
+	go test -tags daemon -v ${PACKAGES}
 
 .PHONY: test-coverage
 test-coverage:
-	./scripts/test/unit-with-coverage
+	( for pkg in ${PACKAGES}; do \
+		go test -tags daemon \
+			-cover \
+			-coverprofile=profile.out \
+			-covermode=atomic \
+			$$pkg || exit;\
+		\
+		if test -f profile.out; then \
+			cat profile.out >> coverage.txt; \
+			rm profile.out; \
+		fi; \
+	done )
```

Comment:
> When would you actually benefit from make not running the coverage because no files are changed? Any test target seems naturally .PHONY to me.

When would you even want your computer to spend time doing unnecessary work when the natural and idiomatic way to use the tool at hand avoids it?

> I'm also not a fan of leaving a bunch of garbage .out files around. Removing them after the concat is the right thing to do.

So add them to `.gitignore` and if they really offend you prepend a `.` on the filename.

> I don't think it's unreasonable to say that most maintainers of this repo are more familiar with bash than they are Makefile syntax, so maintaining bash is going to be a better option 

It's not like make is some arcane little known tool or that the usage here is in any way baroque. Anyone with the ability to contribute to or maintain this project shouldn't have a problem with it on the odd occasion they find the build needs tweaking.